File: frontend/src/types/feedback.ts
import type { ActivityStatus } from './activity'

export interface FeedbackPayload {
  overallRating: number
  contentRating: number
  serviceRating: number
  comment?: string
}

export interface Feedback extends FeedbackPayload {
  id: number
  activityId: number
  activityTitle: string
  createdAt: string
  updatedAt: string
}

export interface FeedbackStatus {
  activityId: number
  activityTitle: string
  activityStatus: ActivityStatus
  required: boolean
  attended: boolean
  canSubmit: boolean
  unavailableReason?: string
  feedbackDeadline?: string
  feedback?: Feedback
}

export interface FeedbackDashboard {
  activityId: number
  activityTitle: string
  registeredCount: number
  attendedCount: number
  feedbackCount: number
  responseRate: number
  averageOverallRating: number
  averageContentRating: number
  averageServiceRating: number
  overallRatingDistribution: Record<number, number>
  feedbacks: Feedback[]
}

File: backend/src/main/java/com/example/activity/entity/Feedback.java
package com.example.activity.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("feedbacks")
public class Feedback {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long activityId;
    private Long userId;
    private Integer overallRating;
    private Integer contentRating;
    private Integer serviceRating;
    private String comment;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
